packages/firestore/src/local/indexeddb_schema_converter.ts

```diff
@@ -401,6 +423,60 @@ export class SchemaConverter implements SimpleDbSchemaConverter {
       return targetStore.put(updatedDbTarget);
     });
   }
+
+  private rewriteRemoteDocumentCache(
+    db: IDBDatabase,
+    transaction: SimpleDbTransaction
+  ): PersistencePromise<void> {
+    const legacyRemoteDocumentStore = transaction.store<
+      DbRemoteDocumentKeyLegacy,
+      DbRemoteDocumentLegacy
+    >(DbRemoteDocumentStoreLegacy);
+
+    const remoteDocumentStore = db.createObjectStore(DbRemoteDocumentStore, {
+      keyPath: DbRemoteDocumentKeyPath
+    });
+    remoteDocumentStore.createIndex(
+      DbRemoteDocumentDocumentKeyIndex,
+      DbRemoteDocumentDocumentKeyIndexPath
+    );
+    remoteDocumentStore.createIndex(
+      DbRemoteDocumentReadTimeIndex,
+      DbRemoteDocumentReadTimeIndexPath,
+      { unique: false }
+    );
+    remoteDocumentStore.createIndex(
+      DbRemoteDocumentCollectionGroupIndex,
+      DbRemoteDocumentCollectionGroupIndexPath
+    );
+
+    const writes: Array<PersistencePromise<void>> = [];
+    return legacyRemoteDocumentStore
+      .iterate((_, legacyDocument) => {
+        const remoteDocumentStore = transaction.store<
```

Comment:
Is this redeclaration of `remoteDocumentStore` necessary?

Reply:
One of them is an IDBObjectStore, the other one is a SimpleDbObjectStore. I moved the object store creation out of this function.